### PR TITLE
This vectorizes TensorPrimitives.Log2

### PR DESCRIPTION
**Pull request by tannergooding**
This makes progress towards https://github.com/dotnet/runtime/issues/92219

### CODE REVIEW

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netstandard.cs [hunk 1]
```diff
@@ -903,24 +904,40 @@ public Vector<float> Invoke(Vector<float> x, Vector<float> y)
...
         private readonly struct AbsoluteOperator : IUnaryOperator
         {
+            public bool CanVectorize => true;
             public float Invoke(float x) => MathF.Abs(x);
             public Vector<float> Invoke(Vector<float> x) => Vector.Abs(x);
         }
 
+        private readonly struct Log2Operator : IUnaryOperator
+        {
+            public bool CanVectorize => false;
```
Comment (stephentoub) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netstandard.cs:928]: Total nit:
```suggestion
            public bool CanVectorize => false;

```

**Review by stephentoub — APPROVED** [review]
Nice.

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
Looks like are some NaN-related test failures

**Comment by tannergooding** [comment]
> Looks like there are some NaN-related test failures

Should be handled now. The issue was that `vrs4_log2f` was deferring the special cases to the scalar `log2f` (which I missed initially): https://github.com/amd/aocl-libm-ose/blob/aocl-4.1/src/optimized/vec/vrs4_log2f.c#L183-L185

The actual logic required to handle them as vectorized isn't terribly complicated, so I ported that: https://github.com/amd/aocl-libm-ose/blob/aocl-4.1/src/optimized/log2f.c#L173-L205

Basically just some additional ternaries in the case where such a value is passed in.

**Comment by tannergooding** [comment]
Will do
